guides/pull-requests.md

```diff
@@ -0,0 +1,40 @@
+---
+explains: How to suggest changes and how to review them
+---
+
+# Pull requests
+
+Pull requests (as they are named on GitHub) or Merge requests (as they are named on GitLab) are proposed changes to a codebase. They are submitted by a contributor to a project and accepted or rejected by a repository's maintainers. Pull requests each have their own discussion forum enabling a collaborative review of the committed code.
+
+## Making a Pull Request
+
+When you make a Pull Request, make sure the code you are suggesting and tidy, as it will be reviewed by the project maintainers line-by-line. 
+
+Try to stay within the scope of what you are trying to fix. PRs with other fixes inside of them—even tiny corrections of typos—often make it hard to effectively review them. If you do want to fix a typo, make a new PR for that. It will surely be accepted and merged in no-time.
+
+Additionally, when making new commits to your branch, try to be clear about what you've done. Give the reviewer an idea of the changes you've introduced, and why. This will also make it easy for future contributors to understand your changes.
+
```

Comment:
As commented earlier to PR #18 

Additionally, is the code affected by the PR logically combined.
For example, pushing new functionality but also adding a typo fix in the same PR, instead of separating this. As a result, making it difficult to cherry-pick PR commits, because the commit can affect files not logically expected from the PR.

Are all commits rebased into 1 single commit and does is contain a meaningfull commit message.

Reply:
The role of these two paragraphs was actually to cover that,

* Clear and scoped PR's
* Good commit history

I am a bit afraid of making this guide too complex for _newcomers_ to understand. Any suggestions on good external guides or how to simplify this?

Reply:
I understand, BUT, failing to address this early can create a lot of mess in the future.

Not rebasing makes your master history a mess. And, experimenting on a fork should be safe, as long as you don't have merge conflicts (that is already too complex for newcomers) and therefor don't have to sync your fork.

Just learn them to do this before every commit to a PR:
```
git commit -m "bla" -m "bla bla bla"
git rebase -i HEAD~2
# fixup bla with the previous commit
# save and close
git push -f origin master
```
That's something everybody should be able to do.

And my team members learn to use `git diff` and `git add -N` extensively to make sure they don't commit things they didn't want to.

Reply:
So where does the team publish their code? I see nothing on github...